src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -8,17 +9,55 @@ def initialize
         set_server_facts
       end
 
+      # Compiles a catalog according to the spec provided from the
+      # request.
+      # @param [Hash] request_data details about the catalog to be compiled
+      # @return [Hash] containing either just the catalog or catalog and logs,
+      #                if capturing logs was enabled
       def compile(request_data)
-        processed_hash = convert_java_args_to_ruby(request_data)
+        options = request_data['options'] || {}
+        # Default to capturing errors and warnings from compiles
+        options['capture_logs'] = true unless options['capture_logs']
```

Comment:
won't this make it always true

Reply:
Oh, shoot, yeah, I needed a `nil?` in there. I think Justin and I had discussed also making this false by default, so this line can be removed.

Reply:
In this kind of case also we should prefer to save the result to a local variable instead of updating the hash. Then you can get from the hash with a default like
```ruby
capture_logs = options.fetch('capture_logs', true)
```

Reply:
Oh I forgot about `fetch`, that's much nicer. Still think it makes sense to have this off by default though.

Reply:
if it's off by default then no